compiler/modules/front.syntax/preparse.bal

```diff
@@ -41,7 +42,7 @@ function preparseBracketedTypeDesc(Tokenizer tok, CLOSE_BRACKET close) returns b
     if t == () {
         return tok.err("incomplete statement");
     }
-    return t != ".";
+    return t is [IDENTIFIER, string] || t == "_" || t == "|" || t == "&" || t =="?"; 
```

Comment:
Why can't we keep this as a negative test?

Reply:
For a negative test we'll also have to consider `=` and compound assignment operators.